strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/providers/io/MavenMetadataExpiredRepositoryPathHandler.java

```diff
@@ -0,0 +1,127 @@
+package org.carlspring.strongbox.providers.io;
+
+import org.carlspring.commons.encryption.EncryptionAlgorithmsEnum;
+import org.carlspring.strongbox.providers.repository.proxied.ProxyRepositoryArtifactResolver;
+import org.carlspring.strongbox.storage.repository.Repository;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.lang.reflect.UndeclaredThrowableException;
+import java.nio.file.Files;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import static org.carlspring.strongbox.providers.io.MavenMetadataExpiredRepositoryPathHandler.Decision.*;
+
+/**
+ * @author Przemyslaw Fusik
+ */
+@Component
+public class MavenMetadataExpiredRepositoryPathHandler
+        implements MavenExpiredRepositoryPathHandler
+{
+
+    private static final Logger logger = LoggerFactory.getLogger(MavenMetadataExpiredRepositoryPathHandler.class);
+
+    @Inject
+    private ProxyRepositoryArtifactResolver proxyRepositoryArtifactResolver;
+
+    @Override
+    public boolean supports(final RepositoryPath repositoryPath)
+    {
+        if (repositoryPath == null)
+        {
+            return false;
+        }
+        try
+        {
+            if (!RepositoryFiles.isMetadata(repositoryPath))
+            {
+                return false;
+            }
+        }
+        catch (IOException e)
+        {
+            throw new UndeclaredThrowableException(e);
+        }
+
+        Repository repository = repositoryPath.getRepository();
+        return repository.getRemoteRepository() != null;
+    }
+
+    @Override
+    public void handleExpiration(final RepositoryPath repositoryPath)
+            throws IOException
+    {
+        Decision refetchMetadata = determineMetadataRefetch(repositoryPath,
+                                                            EncryptionAlgorithmsEnum.SHA1);
+        if (refetchMetadata == I_DONT_KNOW)
```

Comment:
Hahaha!!! :D

Reply:
I think @fuss86 just received +9999 player points for this. :joy: